common/src/main/java/org/apache/rocketmq/common/Configuration.java

```diff
@@ -219,6 +221,35 @@ public void persist() {
         }
     }
 
+    public void persistBrokerConf(Properties from) {
+        FileReader reader = null;
+        try {
+            readWriteLock.readLock().lockInterruptibly();
+            String fileName = this.getStorePath();
+            File file = new File(fileName);
+            if (!file.exists()) {
+                file.createNewFile();
+            }
+            Properties properties = new Properties();
+            reader = new FileReader(fileName);
+            properties.load(reader);
+            merge(from, properties);
+            final String str = MixAll.properties2String(properties);
+            MixAll.string2File(str, fileName);
+        } catch (Exception e) {
+            log.error("persist brokerConf error", e);
+        } finally {
+            readWriteLock.readLock().unlock();
+            if (reader != null) {
+                try {
+                    reader.close();
+                } catch (IOException e) {
+                    log.error("Close FileReader occur error", e);
```

Comment:
unable to close FileReader

Reply:
Thanks for your review, I have solved it